vyper/venom/passes/sccp/sccp.py

```diff
@@ -228,28 +228,35 @@ def _eval(self, inst) -> LatticeItem:
         changed.
         """
         opcode = inst.opcode
-
+        ret = None
         ops = []
         for op in inst.operands:
-            if isinstance(op, IRVariable):
-                ops.append(self.lattice[op])
-            elif isinstance(op, IRLabel):
-                return LatticeEnum.BOTTOM
-            else:
-                ops.append(op)
+            # Evaluate the operand according to the lattice
+            match op:
+                case IRLabel():
+                    return LatticeEnum.BOTTOM
+                case IRVariable():
+                    eval_result = self.lattice[op]
+                case _:
+                    eval_result = op
+
+            # If any operand is BOTTOM, the whole operation is BOTTOM
+            # and we can stop the evaluation early
+            if eval_result is LatticeEnum.BOTTOM:
+                ret = LatticeEnum.BOTTOM
+                break
 
-        ret = None
-        if LatticeEnum.BOTTOM in ops:
-            ret = LatticeEnum.BOTTOM
-        else:
+            ops.append(eval_result)
+
+        # If we haven't found BOTTOM yet, evaluate the operation
+        if ret is None:
             if opcode in ARITHMETIC_OPS:
```

Comment:
we don't need this guard since _eval is already guarded. we can simply `assert opcode in ARITHMETIC_OPS`, and remove the else branch as well.